atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftMemberContext.java

```diff
@@ -418,4 +419,26 @@ public SnapshotChunkReader getSnapshotChunkReader() {
   public void setSnapshotChunkReader(final SnapshotChunkReader snapshotChunkReader) {
     this.snapshotChunkReader = snapshotChunkReader;
   }
+
+  public boolean hasNextEntry() {
+    return reader.hasNext();
+  }
+
+  public Indexed<RaftLogEntry> nextEntry() {
+    currentEntry = reader.next();
+    return currentEntry;
+  }
+
+  public Indexed<RaftLogEntry> getCurrentEntry() {
+    return currentEntry;
+  }
+
+  public long getCurrentIndex() {
+    return currentEntry != null ? currentEntry.index() : 0;
```

Comment:
Would it make sense to return `journal.getFirstIndex() - 1`?

Reply:
I set it this way to be functionally similar to how it worked previously. The Mapped and FileBased Readers implement `getCurrentIndex` like this